chromium_src/components/history/core/browser/sync/typed_url_sync_bridge.cc

```diff
@@ -0,0 +1,60 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "components/history/core/browser/sync/typed_url_sync_bridge.h"
+
+#include "brave/components/brave_sync/features.h"
+#include "ui/base/page_transition_types.h"
+
+namespace {
+
+bool IsSendAllHistoryEnabled() {
+  return base::FeatureList::IsEnabled(
+      brave_sync::features::kBraveSyncSendAllHistory);
+}
+
+}  // namespace
+
+namespace ui {
+
+static bool PageTransitionCoreTypeIs_BraveImpl(PageTransition lhs,
+                                               PageTransition rhs) {
+  if (IsSendAllHistoryEnabled() &&
+      PageTransitionCoreTypeIs(rhs, ui::PAGE_TRANSITION_RELOAD)) {
+    // At TypedURLSyncBridge::WriteToTypedUrlSpecifics when syncing all
+    // history, don't ignore reload transitions.
+    return false;
+  }
```

Comment:
I don't see the benefits of syncing RELOAD, the original record has already synced and available on remote devices.
We need to patch lot of stuff including test because of it. Also I don't understand how it can be useful to see my remote devices saturated with reloads except order which I also don't feel like most users would care so much about.

Reply:
Thanks @darkdh , this is implemented in https://github.com/brave/brave-core/pull/16705/commits/701e9d7530650dfc538aa527774e2ef7260e09b3

And PR became much cleaner, all the upstream tests work well as-is, except `BraveTypedURLSyncBridgeTest.ThrottleVisitLocalTypedUrl` which required only constants override via `chromium_scr`.